db/types/operations/cast.py

```diff
@@ -597,6 +607,58 @@ def _get_timestamp_without_timezone_type_body_map():
 
 
 def _get_money_type_body_map():
+    """
+    Get SQL strings that create various functions for casting different
+    types to money.
+    We allow casting any number type to money, assuming currency is USD.
+    We allow casting any textual type to money, assuming currency is USD
+    and that the type can be cast through a numeric.
+    """
+    default_behavior_source_types = [MONEY]
+    number_types = NUMBER_TYPES
+    textual_types = TEXT_TYPES
+    cast_loss_exception_str = (
+        f"RAISE EXCEPTION '% cannot be cast to {MONEY} as currency symbol is missing', $1;"
+    )
+
+    def _get_number_cast_to_money():
+        return f"""
+        BEGIN
+          RETURN $1::numeric::{MONEY};
+        END;
+        """
+
+    def _get_base_textual_cast_to_money():
+        return f"""
+        DECLARE currency {TEXT};
+        BEGIN
+          SELECT to_char(1, 'L') INTO currency;
+          IF ($1 LIKE '%' || currency) OR ($1 LIKE currency || '%') THEN
```

Comment:
I think using `LEFT($1, 1)=currency OR RIGHT($1, 1)=currency` here would be easier to understand, and also more performant.

Reply:
`LEFT($1, 1)=currency OR RIGHT($1, 1)=currency`, wouldn't work in the case of currencies with more than one character, for example _danish krone_

Reply:
Ah, you're correct.